storage/src/main/java/org/apache/kafka/server/log/internals/LogConfig.java

```diff
@@ -0,0 +1,504 @@
+        this.flushMs = getLong(TopicConfig.FLUSH_MS_CONFIG);
+        this.retentionSize = getLong(TopicConfig.RETENTION_BYTES_CONFIG);
+        this.retentionMs = getLong(TopicConfig.RETENTION_MS_CONFIG);
+        this.maxMessageSize = getInt(TopicConfig.MAX_MESSAGE_BYTES_CONFIG);
+        this.indexInterval = getInt(TopicConfig.INDEX_INTERVAL_BYTES_CONFIG);
+        this.fileDeleteDelayMs = getLong(TopicConfig.FILE_DELETE_DELAY_MS_CONFIG);
+        this.deleteRetentionMs = getLong(TopicConfig.DELETE_RETENTION_MS_CONFIG);
+        this.compactionLagMs = getLong(TopicConfig.MIN_COMPACTION_LAG_MS_CONFIG);
+        this.maxCompactionLagMs = getLong(TopicConfig.MAX_COMPACTION_LAG_MS_CONFIG);
+        this.minCleanableRatio = getDouble(TopicConfig.MIN_CLEANABLE_DIRTY_RATIO_CONFIG);
+        this.compact = getList(TopicConfig.CLEANUP_POLICY_CONFIG).stream()
+            .map(c -> c.toLowerCase(Locale.ROOT))
+            .collect(Collectors.toList())
+            .contains(TopicConfig.CLEANUP_POLICY_COMPACT);
+        this.delete = getList(TopicConfig.CLEANUP_POLICY_CONFIG).stream()
+            .map(c -> c.toLowerCase(Locale.ROOT))
+            .collect(Collectors.toList())
+            .contains(TopicConfig.CLEANUP_POLICY_DELETE);
+        this.uncleanLeaderElectionEnable = getBoolean(TopicConfig.UNCLEAN_LEADER_ELECTION_ENABLE_CONFIG);
+        this.minInSyncReplicas = getInt(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG);
+        this.compressionType = getString(TopicConfig.COMPRESSION_TYPE_CONFIG).toLowerCase(Locale.ROOT);
+        this.preallocate = getBoolean(TopicConfig.PREALLOCATE_CONFIG);
+        this.messageFormatVersion = MetadataVersion.fromVersionString(getString(TopicConfig.MESSAGE_FORMAT_VERSION_CONFIG));
+        this.messageTimestampType = TimestampType.forName(getString(TopicConfig.MESSAGE_TIMESTAMP_TYPE_CONFIG));
+        this.messageTimestampDifferenceMaxMs = getLong(TopicConfig.MESSAGE_TIMESTAMP_DIFFERENCE_MAX_MS_CONFIG);
+        this.leaderReplicationThrottledReplicas = Collections.unmodifiableList(getList(LogConfig.LEADER_REPLICATION_THROTTLED_REPLICAS_CONFIG));
+        this.followerReplicationThrottledReplicas = Collections.unmodifiableList(getList(LogConfig.FOLLOWER_REPLICATION_THROTTLED_REPLICAS_CONFIG));
+        this.messageDownConversionEnable = getBoolean(TopicConfig.MESSAGE_DOWNCONVERSION_ENABLE_CONFIG);
+
+        remoteLogConfig = new RemoteLogConfig(this, retentionMs, retentionSize);
+    }
+
+    @SuppressWarnings("deprecation")
+    public RecordVersion recordVersion() {
+        return messageFormatVersion.highestSupportedRecordVersion();
+    }
+
+    // Exposed as a method so it can be mocked
+    public int maxMessageSize() {
+        return maxMessageSize;
+    }
+
+    public long randomSegmentJitter() {
+        if (segmentJitterMs == 0)
+            return 0;
+        else
+            return Utils.abs(ThreadLocalRandom.current().nextInt()) % Math.min(segmentJitterMs, segmentMs);
```

Comment:
Slight change here: we now use a `ThreadLocalRandom` instead of a global `Random`.